taskcluster/ac_taskgraph/release_promotion.py

```diff
@@ -0,0 +1,172 @@
+        project=parameters['project']
+    )
+    rebuild_kinds = input.get('rebuild_kinds') or promotion_config.get('rebuild-kinds', [])
+    do_not_optimize = input.get('do_not_optimize') or promotion_config.get('do-not-optimize', [])
+
+    # make parameters read-write
+    parameters = dict(parameters)
+    # Build previous_graph_ids from ``previous_graph_ids`` or ``revision``.
+    previous_graph_ids = input.get('previous_graph_ids')
+    if not previous_graph_ids:
+        previous_graph_ids = [find_decision_task(parameters, graph_config)]
+
+    # Download parameters from the first decision task
+    parameters = get_artifact(previous_graph_ids[0], "public/parameters.yml")
+    # Download and combine full task graphs from each of the previous_graph_ids.
+    # Sometimes previous relpro action tasks will add tasks, like partials,
+    # that didn't exist in the first full_task_graph, so combining them is
+    # important. The rightmost graph should take precedence in the case of
+    # conflicts.
+    combined_full_task_graph = {}
+    for graph_id in previous_graph_ids:
+        full_task_graph = get_artifact(graph_id, "public/full-task-graph.json")
+        combined_full_task_graph.update(full_task_graph)
+    _, combined_full_task_graph = TaskGraph.from_json(combined_full_task_graph)
+    parameters['existing_tasks'] = find_existing_tasks_from_previous_kinds(
+        combined_full_task_graph, previous_graph_ids, rebuild_kinds
+    )
+    parameters['do_not_optimize'] = do_not_optimize
+    parameters['target_tasks_method'] = target_tasks_method
+    parameters['build_number'] = int(input['build_number'])
+    # When doing staging releases on try, we still want to re-use tasks from
+    # previous graphs.
+    parameters['optimize_target_tasks'] = True
+    parameters['shipping_phase'] = input['release_promotion_flavor']
+
+    version_in_file = read_version_file()
+    parameters['version'] = input['version'] if input.get('version') else read_version_file()
+    version_string = parameters['version']
+    if version_string != version_in_file:
+        raise ValueError("Version given in tag ({}) does not match the one in version.txt ({})".format(version_string, version_in_file))
+    parameters['head_tag'] = 'v{}'.format(version_string)
+
+    parameters['next_version'] = input['next_version']
+
+    version = MavenVersion.parse(version_string)
+    if version.is_beta:
+        release_type = "beta"
+    elif version.is_release:
+        release_type = "release"
+    else:
+        raise ValueError("Unsupported version type: {}".format(version.version_type))
+    parameters['release_type'] = release_type
+
+    parameters['pull_request_number'] = None
+
+    # make parameters read-only
+    parameters = Parameters(**parameters)
+
+    taskgraph_decision({'root': graph_config.root_dir}, parameters=parameters)
+
+
+def read_version_file():
+    with open(os.path.join(os.path.dirname(__file__), '..', '..', 'version.txt')) as f:
```

Comment:
Bug: This file doesn't exist yet. We'll have to create it and to remove this line:

https://github.com/mozilla-mobile/android-components/blob/35c99f4ba90532ec301df4148ac49be48b902004/.buildconfig.yml#L1

Some things are depending on `.buildconfig.yml` so we'll have to update calls to use `version.txt` instead. @escapewindow asked whether we should use `.buildconfig.yml` or `version.txt`. I believe the latter is best. For more context: see https://jira.mozilla.com/browse/RELENG-152?focusedCommentId=197982&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-197982

Reply:
So, if I'm following this correctly is the issue [146](https://jira.mozilla.com/browse/RELENG-146) that adds `version.txt` a blocker to landing this??

Reply:
I'm sorry, I may not follow what you mean regarding [146](https://jira.mozilla.com/browse/RELENG-146). My apologies if my answer doesn't address your point.

I see 2 ways of solving this issue:
 1. either we add `version.txt` and stop using the value in `.buildconfig.yml` 
 2. or we update `read_version_file()` to use `.buildconfig.yml` now , then we do point 1 in a follow up.

I don't have any preference.

Reply:
Aki said he will be adding `version.text` in a separate pr so we'll coordinate the merging of his first.

Reply:
I'm sorry I missed your last message.

Awesome! I see Aki already landed the patch, so we're all good here! Thank you for the context 🙂